src/openrct2/world/Map.cpp

```diff
@@ -1312,14 +1312,11 @@ bool map_can_construct_with_clear_at(
     loc_68B9B7:
         if (gParkFlags & PARK_FLAGS_FORBID_HIGH_CONSTRUCTION)
         {
-            al = pos.clearanceZ / 8 - tileElement->base_height;
-            if (al >= 0)
+            northZ = pos.clearanceZ - tileElement->GetBaseZ();
+            if (northZ >= 0 && northZ > (18 * 8))
```

Comment:
```suggestion
            if (northZ >= 0 && northZ > (18 * COORDS_Z_STEP))
```

Reply:
Also this instance of variable al should be renamed to `heightFromGround`. I was making that change anyway in the #10501

Reply:
Ok, did both.